commands/curation/curationaudit.go

```diff
@@ -461,6 +484,136 @@ func (ca *CurationAuditCommand) auditTree(tech techutils.Technology, results map
 	return err
 }
 
+func getSelectedPackages(requestedRows string, blockedPackages []*PackageStatus) (selectedPackages []*PackageStatus, ok bool) {
+	validFormat := regexp.MustCompile(`^(all|(\d+(-\d+)?)(,\d+(-\d+)?)*$)`)
+	if !validFormat.MatchString(requestedRows) {
+		fmt.Print("Invalid request format.\n\n")
+		return nil, false
+	}
+
+	if requestedRows == "all" {
+		return blockedPackages, true
+	}
+
+	var indices []int
+	parts := strings.Split(requestedRows, ",")
+	for _, part := range parts {
+		if strings.Contains(part, "-") {
+			rangeParts := strings.Split(part, "-")
+			start, _ := strconv.Atoi(rangeParts[0])
+			end, _ := strconv.Atoi(rangeParts[1])
+			for i := start; i <= end; i++ {
+				if slices.Contains(indices, i) {
+					continue
+				}
+				indices = append(indices, i)
+			}
+		} else {
+			index, err := strconv.Atoi(part)
+			if err != nil || slices.Contains(indices, index) {
+				continue
+			}
+			indices = append(indices, index)
+		}
+	}
```

Comment:
```suggestion
	rows := strings.Split(requestedRows, ",")
	for _, rowStr := range parts {
		if strings.Contains(rowStr, "-") {
            // Range of rows requested
			rangeParts := strings.Split(rowStr, "-")
			startRow, _ := strconv.Atoi(rangeParts[0])
			endRow, _ := strconv.Atoi(rangeParts[1])
			for i := start; i <= end; i++ {
				if slices.Contains(indices, i) {
					continue
				}
				indices = append(indices, i)
			}
		} else {
            // Single row requested
			index, err := strconv.Atoi(rowStr)
			if err != nil || slices.Contains(indices, index) {
				continue
			}
			indices = append(indices, index)
		}
	}
```
rename for readability, maybe also refactor this to a separated method
add some comments